feat(radar): add onClick handler

Nino3103 opened the pull request:
## Radar Chart : add onClick handler

- added onClick props in Radar props
- adjusted the types to accept onClick in Radar props
- added a story in Radar storybook with onClick

## Issue : https://github.com/plouc/nivo/issues/1003

## Video
https://github.com/plouc/nivo/assets/48714971/edc20b8a-df04-4cdb-85d2-49efb8b8ce90

charles-glanceable commented:
We are really looking forward to this new feature! Thank you 👍

plouc reviewed (changes requested):
Overall, LGTM, thank you.

However the property should be added to the website (I'm not sure we need a story), also it would be good to add some tests for this new feature.

Nino3103 commented:
Thank you for your answer, I will add it to the website and add some tests 👍 Should I remove the story ?

Nino3103 commented:
Hello @plouc , I added a radar test and updated the website


https://github.com/plouc/nivo/assets/48714971/d5653a24-3606-4681-9707-d077aa178fa1

plouc commented on packages/radar/tests/Radar.test.tsx:
```diff
@@ -222,3 +224,22 @@ describe('legend', () => {
+describe('interactivity', () => {
+    it('should support onClick handler', async () => {
+        const onClick = jest.fn()
+        // const wrapper = mount(<Radar<TestDatum> {...baseProps} />)
```
Should be removed :)

Nino3103 replied:
I removed the commented line (231) :)

plouc commented on storybook/stories/radar/Radar.stories.tsx:
```diff
@@ -227,3 +227,14 @@ export const WithPatterns: Story = {
 export const WithRotation: Story = {
     render: args => <Radar {...commonProperties} curve={args.curve} rotation={36} />,
 }
+
```
I think it can be removed as it's now included to the website, less to maintain :)

Nino3103 commented:
@plouc I've resync with master, tests are passing now 👍

plouc commented:
@Nino3103, thank you for your contribution!
